src/bench/bench.h

```diff
@@ -9,136 +9,48 @@
 #include <map>
 #include <string>
 #include <vector>
-#include <chrono>
 
+#include <bench/nanobench.h>
 #include <boost/preprocessor/cat.hpp>
 #include <boost/preprocessor/stringize.hpp>
 
 struct RegTestingSetup;
 extern const RegTestingSetup* g_testing_setup; //!< A pointer to the current testing setup
-
-// Simple micro-benchmarking framework; API mostly matches a subset of the Google Benchmark
-// framework (see https://github.com/google/benchmark)
-// Why not use the Google Benchmark framework? Because adding Yet Another Dependency
-// (that uses cmake as its build system and has lots of features we don't need) isn't
-// worth it.
-
 /*
  * Usage:
 
-static void CODE_TO_TIME(benchmark::State& state)
+static void CODE_TO_TIME(benchmark::Bench& bench)
 {
     ... do any setup needed...
-    while (state.KeepRunning()) {
+    nanobench::Config().run([&] {
        ... do stuff you want to time...
-    }
+    });
     ... do any cleanup needed...
 }
 
-// default to running benchmark for 5000 iterations
-BENCHMARK(CODE_TO_TIME, 5000);
+BENCHMARK(CODE_TO_TIME);
 
  */
 
 namespace benchmark {
-// In case high_resolution_clock is steady, prefer that, otherwise use steady_clock.
-struct best_clock {
-    using hi_res_clock = std::chrono::high_resolution_clock;
-    using steady_clock = std::chrono::steady_clock;
-    using type = std::conditional<hi_res_clock::is_steady, hi_res_clock, steady_clock>::type;
-};
-using clock = best_clock::type;
-using time_point = clock::time_point;
-using duration = clock::duration;
-
-class Printer;
-
-class State
-{
-public:
-    std::string m_name;
-    uint64_t m_num_iters_left;
-    const uint64_t m_num_iters;
-    const uint64_t m_num_evals;
-    std::vector<double> m_elapsed_results;
-    time_point m_start_time;
 
-    bool UpdateTimer(time_point finish_time);
+using namespace ankerl::nanobench;
```

Comment:
nit: How about:
```cpp
namespace nanobench { using namespace ankerl::nanobench }
```
So that all the external nanobench members are more explicitly identified?

Reply:
Sure I can do that. Would you do that inside the benchmark namespace? then all benchmark arguments would become e.g. 
```cpp
static void Base58Encode(benchmark::nanobench::Bench& bench)
```
Which is a bit long.

Reply:
How about doing individual assignments for the classes in use, e.g.:

```cpp
namespace benchmark { using ankerl::nanobench::Bench }
```

Reply:
I think that's better, `Bench` is the only thing that's needed in the benchmarks anyway. I've commited c2e924f which does that